clients/client/src/test/java/org/projectnessie/model/TestModelObjectsSerialization.java

```diff
@@ -0,0 +1,202 @@
+                        .build())
+                .hasMore(true)
+                .build(),
+            LogResponse.class,
+            "{\"hasMore\":true,"
+                + "\"token\":\""
+                + HASH
+                + "\","
+                + "\"operations\":["
+                + "{\"hash\":\""
+                + HASH
+                + "\","
+                + "\"committer\":\"committer@testnessie.com\",\"author\":\"author@testnessie.com\","
+                + "\"signedOffBy\":\"signer@testnessie.com\",\"message\":\"test commit\","
+                + "\"commitTime\":\""
+                + now
+                + "\","
+                + "\"authorTime\":\""
+                + now
+                + "\","
+                + "\"properties\":{\"prop1\":\"val1\"}}]}"),
+        new Case(
+            ImmutableMerge.builder().fromHash(HASH).fromRefName(branchName).build(),
+            Merge.class,
+            "{\"fromHash\":\"" + HASH + "\"," + "\"fromRefName\":\"testBranch\"}"));
+  }
+
+  static List<Case> negativeCases() {
+    return Arrays.asList(
+        // Special chars in the branch name make it invalid
+        new Case(
+            Branch.class,
+            "{\"type\":\"BRANCH\",\"name\":\"$p@c!@L\"," + "\"hash\":\"" + HASH + "\"}"),
+
+        // Invalid hash
+        new Case(
+            Branch.class,
+            "{\"type\":\"BRANCH\",\"name\":\"testBranch\"," + "\"hash\":\"invalidhash\"}"),
+
+        // No name
+        new Case(
+            Branch.class, "{\"type\":\"BRANCH\",\"name\":null," + "\"hash\":\"" + HASH + "\"}"),
+        new Case(
+            Transplant.class,
+            "{\"hashesToTransplant\":" + "[\"invalidhash\"]," + "\"fromRefName\":null}"),
+
+        // Invalid hash
+        new Case(
+            Transplant.class,
+            "{\"hashesToTransplant\":" + "[\"invalidhash\"]," + "\"fromRefName\":\"testBranch\"}"),
+        new Case(Tag.class, "{\"type\":\"TAG\",\"name\":\"tagname\",\"hash\":\"invalidhash\"}"),
+        new Case(
+            Hash.class, "{\"type\":\"HASH\",\"name\":\"invalidhash\",\"hash\":\"invalidhash\"}"));
+  }
+
+  static class Case {
+
+    final Object obj;
+    final Class<?> deserializeAs;
+    final String deserializedJson;
+
+    public Case(Class<?> deserializeAs, String deserializedJson) {
+      this(null, deserializeAs, deserializedJson);
+    }
+
+    public Case(Object obj, Class<?> deserializeAs, String deserializedJson) {
+      this.obj = obj;
+      this.deserializeAs = deserializeAs;
+      this.deserializedJson = deserializedJson;
+    }
+
+    @Override
+    public String toString() {
+      return deserializeAs.getName() + " : " + obj;
+    }
+  }
+
+  @Test
+  public void testN() throws IOException {
+    Tag t = ImmutableTag.builder().hash(HASH).name("tagname").build();
+    System.out.println(MAPPER.writeValueAsString(t));
+    Hash h = ImmutableHash.builder().name(HASH).build();
+    System.out.println(MAPPER.writeValueAsString(h));
```

Comment:
What is the purpose of this test? Is it perhaps a leftover from some local experiments?

Reply:
Yea - let's not spam the console with non-test code.